Cabal/Distribution/Simple/Utils/Json.hs

```diff
@@ -0,0 +1,34 @@
+module Distribution.Simple.Utils.Json
+    ( Json(..)
+    , renderJson
+    ) where
+
+data Json = JsonArray [Json]
+          | JsonBool !Bool
+          | JsonNull
+          | JsonNumber !Int
+          | JsonObject [(String, Json)]
+          | JsonString !String
+
+renderJson :: Json -> ShowS
+renderJson (JsonArray objs)   =
+  surround "[" "]" $ intercalate "," $ map renderJson objs
+renderJson (JsonBool True)    = showString "true"
+renderJson (JsonBool False)   = showString "false"
+renderJson  JsonNull          = showString "null"
+renderJson (JsonNumber n)     = shows n
+renderJson (JsonObject attrs) =
+  surround "{" "}" $ intercalate "," $ map render attrs
+  where
+    render (k,v) = (surround "\"" "\"" $ showString k) . showString ":" . renderJson v
+renderJson (JsonString s)     = surround "\"" "\"" $ showString s
+
+surround :: String -> String -> ShowS -> ShowS
+surround begin end middle = showString begin . middle . showString end
+
+intercalate :: String -> [ShowS] -> ShowS
+intercalate sep = go
+  where
+    go []     = id
+    go [x]    = x
+    go (x:xs) = x . showString sep . go xs
```

Comment:
(Just a casual observer, not a contributor) I assume the intention is to avoid adding another dependency to Cabal, but this still feels like a candidate for a separate library, even if it is a small library that is a minimal subset of Aeson... cabal-aeson-lite or something. I am sure others have already thought this through though..

Reply:
Since I wrote that comment I ended up releasing an extended version of my llightweight JSON library as https://hackage.haskell.org/package/microaeson as I ended up needing a dependency-minimal JSON library...

In any case, this PR ought to get rid of the redundant JSON impl in favor of the one we're using already for `plan.json`.